### PR TITLE
EDD-49 adds clientId to metrics logger

### PR DESCRIPTION
**Pull request by daniel-zamora**
# Overview

### What is the feature?

Adds clientId to metricsLogger logging where appropriate. 

### What is the Solution?

Added a column to downloads in the database for clientId, so when we go to log a downloadId we will have a clientId as well.

### What areas of the application does this impact?

sqllite migration
pause downloads
resume downloads
finish download
cancel downloads
initiate download
send to eula
send to login
opening deeplinks

# Testing

### Reproduction steps

-dev metricsLogger sends to uat edsc aws cloudwatch

Verify the metrics logger log statements for the following also log client id:
pausing downloads
resuming downloads
finish download
canceling downloads
initiateting download
download that requires eula
download that requires login

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have bumped the `version` field in package.json and ran `npm install`

### CODE REVIEW

File: src/main/utils/initializeDownload.ts [hunk 1]
```diff
@@ -44,11 +44,15 @@ const initializeDownload = async ({
...
+    const metricIds = await processDownloadIdsForMetrics({
+      database,
+      downloadIds
+    })
     metricsLogger({
       eventType: 'DownloadStarted',
       data: {
-        downloadIds: metricIds
+        downloadIds: metricIds,
+        downloadCount: metricIds.length
```
Comment (eudoroolivares2016) [inline src/main/utils/initializeDownload.ts:55]: Do we need this `downloadCount` in the log?

File: src/main/utils/getClientIdWithDownloadId.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    return {
+      clientId,
+      downloadId: metricId
+    }
+  } catch (error) {
+    console.log(`Error in getClientIdWithDownloadId called with downloadId: ${downloadId}, error: ${error}`)
+
+    return {
+      clientId: null,
```
Comment (eudoroolivares2016) [inline src/main/utils/getClientIdWithDownloadId.ts:23]: Do the metrics on cloud-watch/splunk don't have this field printing in cases where the client-id is null? (We could test this by running EDSC locally starting a download and removing the client-id being passed from the SC)

File: src/main/utils/initializeDownload.ts (hunk 1 above)
Comment (macrouch) [reply src/main/utils/initializeDownload.ts:55]: Is there a reason we would want to track downloads created separately than downloads started? If we just wanted to know that downloads were created and which client sent the download to EDD we could log that metric in `openUrl.ts` and not have to save the clientId in the database and do extra calls here to retrieve the information

**Review by eudoroolivares2016 — APPROVED** [review]

File: src/main/eventHandlers/__tests__/openUrl.test.ts [hunk 3]
```diff
@@ -55,6 +61,15 @@ describe('openUrl', () => {
         }
       )
 
+      expect(metricsLogger).toHaveBeenCalledTimes(1)
+      expect(metricsLogger).toHaveBeenCalledWith({
+        eventType: 'OpenUrl',
+        data: {
+          clientId: 'eed-edsc-dev-serverless-client',
```
Comment (eudoroolivares2016) [inline src/main/eventHandlers/__tests__/openUrl.test.ts:68]: Consider adding a test-case if the client-id is null

File: src/main/eventHandlers/__tests__/openUrl.test.ts [hunk 4]
```diff
@@ -64,7 +79,7 @@ describe('openUrl', () => {
 
     test('does not call startPendingDownloads when an update is available', async () => {
       const appWindow = {}
-      const deepLink = 'earthdata-download://startDownload?getLinks=http%3A%2F%2Flocalhost%3A3000%2Fgranule_links%3Fid%3D42%26flattenLinks%3Dtrue%26linkTypes%3Ddata&downloadId=shortName_versionId&token=Bearer mock-token'
+      const deepLink = 'earthdata-download://startDownload?getLinks=http%3A%2F%2Flocalhost%3A3000%2Fgranule_links%3Fid%3D42%26flattenLinks%3Dtrue%26linkTypes%3Ddata&downloadId=shortName_versionId&clientId=eed-edsc-dev-serverless-client&token=Bearer mock-token'
```
Comment (dpesall) [inline src/main/eventHandlers/__tests__/openUrl.test.ts:82]: We should probably leave the `clientId` out of the deep links for one of these tests to show that we expect to get

`data: {
          clientId: null,
          downloadId: 'shortName_versionId'
}`

if there's no `clientId` found in the deep link.

**Review by dpesall — APPROVED** [review]
